streaming/format/base/phase.py

```diff
@@ -0,0 +1,233 @@
+                f'StreamingDataset or Stream argument `download_max_size` or set it to `None` ' +
+                f'to disable this check completely.')
+
+    def get_local_filename(self) -> str:
+        """Get this phase's local filename.
+
+        Returns:
+            str: Local filename.
+        """
+        return os.path.join(self.stream.local, self.stream.split or '', self.relative_path)
+
+    def get_remote_filename(self) -> str:
+        """Get this phase's remote filename.
+
+        Returns:
+            str: Remote filename or path.
+        """
+        if not self.stream.remote:
+            raise ValueError('Asked for a remote path, but remote does not exist.')
+        return os.path.join(self.stream.remote, self.stream.split or '', self.relative_path)
+
+    def probe(self, listing: Set[str]) -> bool:
+        """Probe the given directory listing for this file phase.
+
+        Try to minimize hitting the filesystem, for performance resaons.
+
+        Args:
+            listing (Set[str]): Recursive dataset file listing.
+
+        Returns:
+            bool: Whether this phase is present or not in the listing.
+        """
+        filename = self.get_local_filename()
+        return filename in listing
+
+    def inventory_local(self, listing: Set[str]) -> Optional[int]:
+        """Initialize the given local directory wrf this shard file phase.
+
+        Args:
+            listing (Set[str]): Recursive dataset file listing.
+
+        Returns:
+            int: Disk usage.
+        """
+        filename = self.get_local_filename()
+
+        # Is the file in the listing?
+        if filename not in listing:
+            # Problem: it's not listed.
+            got_size = None
+        else:
+            # It's listed. Is the file actually there in the filesystem?
+            if not os.path.exists(filename):
+                # Problem: it's not there anymore.
+                got_size = None
+            elif not os.path.isfile(filename):
+                # Problem: it's not a file.
+                rmtree(filename)
+                got_size = None
+            else:
+                # It's there. Do we know its expected size?
+                if self.size is not None:
+                    # Expected size is known, so they had better match.
+                    got_size = os.stat(filename).st_size
+                    if got_size != self.size:
+                        os.remove(filename)
+                        got_size = None
+                else:
+                    # Expected size is unknown. Is that a problem?
+                    if self.stream.allow_unchecked_resumption:
+                        # Not a problem.
+                        got_size = os.stat(filename).st_size
+                    else:
+                        # Is a problem.
+                        os.remove(filename)
+                        got_size = None
+
+        self.got_size = got_size
```

Comment:
nit: can you change this to `size` instead of `got_size`

Reply:
https://github.com/mosaicml/streaming/pull/585/commits/21f06e0e1f35e85b8e1be63e14422d4d87ca640b